tiledb/sm/array_schema/domain.cc

```diff
@@ -595,12 +586,14 @@ const ByteVecValue& Domain::tile_extent(unsigned i) const {
   return dimension_ptrs_[i]->tile_extent();
 }
 
-std::vector<ByteVecValue> Domain::tile_extents() const {
-  std::vector<ByteVecValue> ret(dim_num_);
-  for (unsigned d = 0; d < dim_num_; ++d)
+tdb::pmr::vector<ByteVecValue> Domain::tile_extents() const {
+  tdb::pmr::vector<ByteVecValue> ret(
+      dim_num_, memory_tracker_->get_resource(MemoryType::DOMAINS));
+  for (unsigned d = 0; d < dim_num_; ++d) {
     ret[d] = tile_extent(d);
+  }
 
-  return ret;
+  return {ret, memory_tracker_->get_resource(MemoryType::DOMAINS)};
```

Comment:
This is probably because I removed the move constructor. I think we're probably going to end up undoing the forbidding of move construction, while still preventing move assignment since the later is where we get the implicit copies and undefined behavior.